20_aws_networking/variables.tf

```diff
@@ -0,0 +1,316 @@
+      size                  = number
+      type                  = string
+      delete_on_termination = bool
+    })
+  })
+}
+
+variable "nat_asg_config" {
+  type = object({
+    name                      = string
+    desired_capacity          = number
+    max_size                  = number
+    min_size                  = number
+    launch_template_version   = string
+    health_check_type         = string
+    health_check_grace_period = number
+    tags                      = map(string)
+  })
+  description = "The configuration for the NAT auto scaling group"
+  default = {
+    name                      = "nat-asg"
+    desired_capacity          = 1
+    max_size                  = 2
+    min_size                  = 1
+    launch_template_version   = "$Latest"
+    health_check_type         = "EC2"
+    health_check_grace_period = 300
+    tags = {
+      Name = "nat-asg-instance"
+    }
+  }
+}
+
+
+variable "nat_sg_settings" {
+  description = "The settings for the NAT security group"
+  type = object({
+    name = string
+    ingress = map(object({
+      protocol   = string
+      from_port  = number
+      to_port    = number
+      cidr_block = string
+    }))
+    egress = map(object({
+      protocol   = string
+      from_port  = number
+      to_port    = number
+      cidr_block = string
+    }))
+  })
+  default = {
+    name = "nat-sg"
+    ingress = {
+      default = {
+        protocol   = "-1"
+        from_port  = 0
+        to_port    = 0
+        cidr_block = ""
+      }
+    }
+    egress = {
+      default = {
+        protocol   = "-1"
+        from_port  = 0
+        to_port    = 0
+        cidr_block = "0.0.0.0/0"
+      }
+    }
+  }
+}
+
+variable "public_sg_settings" {
+  description = "The settings for the public security group"
+  type = object({
+    name = string
+    ingress = map(object({
+      from_port  = number
+      to_port    = number
+      protocol   = string
+      cidr_block = string
+    }))
+    egress = map(object({
+      protocol   = string
+      cidr_block = string
+    }))
+  })
+  default = {
+    name = "public-sg"
+    ingress = {
+      http = {
+        from_port  = 80
+        to_port    = 80
+        protocol   = "tcp"
+        cidr_block = "0.0.0.0/0"
+      }
+    }
+    egress = {
+      default = {
+        protocol   = "-1"
+        cidr_block = "0.0.0.0/0"
+      }
+    }
+  }
+}
+
+variable "private_sg_settings" {
+  description = "The settings for the private security group"
+  type = object({
+    name = string
+    ingress = map(object({
+      from_port  = number
+      to_port    = number
+      protocol   = string
+      cidr_block = string
+    }))
+    egress = map(object({
+      protocol   = string
+      cidr_block = string
+    }))
+  })
+  default = {
+    name = "private-sg"
+    ingress = {
+      icmp = {
+        from_port  = -1
+        to_port    = -1
+        protocol   = "icmp"
+        cidr_block = ""
+      }
+      https = {
```

Comment:
Naming problem. If you will need one more https rule with both cidr and source security group id .